### PR TITLE
Optimize token balances in Rest API

### LINKED ISSUE #4131: Optimize token balances in REST API

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

**Related issue(s)**:

Fixes #4131 

**Notes for reviewer**:

Had to disable v2 schema rest api workflow since citus doesn't support such query: `correlated subqueries are not supported when the FROM clause contains a CTE or subquery`

k6 report:

- `main` branch

| URL | VUS | Pass% | RPS | Pass RPS | Avg. Req Duration | Comment |
|----------|-----|-------|-----|----------|-------------------|---------|
| /accounts/{id}/allowances/crypto | 500 | 100.00 | 3116.68/s | 3116.68/s | 159.76ms | |
| /accounts/{id}/allowances/crypto | 500 | 100.00 | 3072.10/s | 3072.10/s | 162.07ms | |
| /accounts | 500 | 100.00 | 378.55/s | 378.55/s | 858.24ms | |
| /accounts?balance=false | 500 | 100.00 | 1483.73/s | 1483.73/s | 246.78ms | |
| /accounts?balance=false&account.publickey={publicKey} | 500 | 100.00 | 2484.76/s | 2484.76/s | 200.39ms | |
| /accounts?account.balance=gt:0 | 500 | 100.00 | 1341.87/s | 1341.87/s | 256.70ms | |
| /accounts?account.balance=ne:{balance}&order=desc | 500 | 100.00 | 1343.04/s | 1343.04/s | 259.54ms | |
| /accounts/{accountId} | 500 | 100.00 | 2290.16/s | 2290.16/s | 217.42ms | |
| /accounts?account.id=ne:{accountId}&order=desc | 500 | 100.00 | 1464.15/s | 1464.15/s | 243.06ms | |
| /balances | 500 | 100.00 | 1582.48/s | 1582.48/s | 267.06ms | |

- PR branch

| URL | VUS | Pass% | RPS | Pass RPS | Avg. Req Duration | Comment |
|----------|-----|-------|-----|----------|-------------------|---------|
| /accounts/{id}/allowances/crypto | 500 | 100.00 | 3133.02/s | 3133.02/s | 158.94ms | |
| /accounts/{id}/allowances/crypto | 500 | 100.00 | 3091.40/s | 3091.40/s | 161.02ms | |
| /accounts | 500 | 100.00 | 379.27/s | 379.27/s | 866.80ms | |
| /accounts?balance=false | 500 | 100.00 | 1476.24/s | 1476.24/s | 245.94ms | |
| /accounts?balance=false&account.publickey={publicKey} | 500 | 100.00 | 2516.38/s | 2516.38/s | 197.85ms | |
| /accounts?account.balance=gt:0 | 500 | 100.00 | 1342.99/s | 1342.99/s | 251.94ms | |
| /accounts?account.balance=ne:{balance}&order=desc | 500 | 100.00 | 1336.33/s | 1336.33/s | 260.16ms | |
| /accounts/{accountId} | 500 | 100.00 | 2387.00/s | 2387.00/s | 208.69ms | |
| /accounts?account.id=ne:{accountId}&order=desc | 500 | 100.00 | 1460.37/s | 1460.37/s | 247.47ms | |
| /balances | 500 | 100.00 | 1593.69/s | 1593.69/s | 263.34ms | |

There is no noticeable performance difference. 

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-rest/config/application.yml [hunk 1]
```diff
@@ -109,6 +109,9 @@ hedera:
         limit:
           default: 25
           max: 100
+          tokenBalance:
+            multiple: 1000
+            single: 2000
```
Comment (xin-hedera) [inline hedera-mirror-rest/config/application.yml:114]: open to better config names

Comment (steven-sheehy) [inline hedera-mirror-rest/config/application.yml:114]:
```suggestion
            multiple: 50
            single: 1000
```

Comment (steven-sheehy) [reply hedera-mirror-rest/config/application.yml:114]: Maybe just add `Accounts`/`Account` to the end so it clarifies it's no the tokenBalance that is multiple/single?

File: hedera-mirror-rest/__tests__/specs/accounts/limit-capped-at-max.json [hunk 2]
```diff
@@ -101,14 +101,16 @@
...
-            "token_realm": 0,
             "token_num": 100002,
             "balance": 55
+          },
+          {
+            "token_num": 100003,
+            "balance": 55
           }
         ]
       },
```
Comment (steven-sheehy) [inline hedera-mirror-rest/__tests__/specs/accounts/limit-capped-at-max.json:116]: Why does this spec have a next link if the number of accounts in the response (6) is less than limit (10)?

**Review by steven-sheehy — CHANGES_REQUESTED** [review]
Also curious about the k6 test run to verify no performance regression.

File: hedera-mirror-rest/__tests__/specs/accounts/limit-capped-at-max.json (hunk 2 above)
Comment (xin-hedera) [reply hedera-mirror-rest/__tests__/specs/accounts/limit-capped-at-max.json:116]: the config is overridden in the spec:

```json
  "setup": {
    "config": {
      "response": {
        "limit": {
          "default": 3,
          "max": 6,
          "tokenBalance": {
            "multiple": 2
          }
        }
      }
    }
```

the `max` is set to 6.

File: hedera-mirror-rest/config/application.yml (hunk 1 above)
Comment (xin-hedera) [reply hedera-mirror-rest/config/application.yml:114]: sure

**Review by steven-sheehy — APPROVED** [review]
LGTM pending performance numbers

**Review by edwin-greene — APPROVED** [review]
Looks good

**Pull request merged**